warehouse/client/controlplane/client.go

```diff
@@ -89,6 +126,20 @@ func NewInternalClientWithCache(baseURI string, auth BasicAuth) InternalControlP
 	}
 }
 
+func (cc *internalClientWithCache) GetSSHKeys(ctx context.Context, id string) (*PublicPrivateKeyPair, error) {
+	if val, ok := cc.cache.Load(id); ok {
```

Comment:
Is it ok for `GetSSHKeys` ids and `GetDestinationSSHKeys` to be stored in the same cache?

In other words, will id_1 returned by `SSHKeys` always represent the same content as id_1 from `DestinationSSHKeys`

Reply:
The id's are different so there won't be any clash. I can move to a different cache but the chance of collision isn't very likely and we will be removing the cache lookup using destination_id in next release. 

Let me know what you think ?

Reply:
We can also prefix to avoid collisions, if suid is used it should be ok. If we plan to remove this code in the future I am ok keeping it like this.

Reply:
We will be removing the destination ssh keys functionality in future, so for now it seems okay. I'll track the corresponding changes in another ticket for now.